src/customcamera.py

```diff
@@ -0,0 +1,83 @@
+from os import getcwd
+
+from picar.SunFounder_PCA9685 import Servo
+from picar import filedb
+
+
+'''
+Camera class.
+
+Uses SunFounder Servo class to control camera servos allowing for pan and tilt.
+Uses SunFounder filedb to configure offset for pan and tilt functions.
+'''
+class Camera:
+    pan_channel = 1
+    tilt_channel = 2
+
+
+    def __init__(self, bus_number=1):
+        self.db = filedb.fileDB(getcwd() + "/../SunFounder_PiCar-V/remote_control/remote_control/driver/config")
+        self.pan_offset = int(self.db.get('pan_offset', default_value=0))
+        self.tilt_offset = int(self.db.get('tilt_offset', default_value=0))
+        self.pan_servo = Servo.Servo(self.pan_channel, bus_number=bus_number, offset=self.pan_offset)
+        self.tilt_servo = Servo.Servo(self.tilt_channel, bus_number=bus_number, offset=self.tilt_offset)
+
+
+    '''
+    Controls the horizontal movement of the camera.Usable inputs are [0,180]. 0 refers to the 
+    vehicles righthand side while 180 refers to the vehicles lefthand side. 
+    '''
+    def pan(self, degree):
+        self.pan_servo.write(self.limit_pan(degree))
+
+
+    '''
+    Writes pan offset to a config file within current directory, lower values direct the 
+    camera more toward the vehicles righthand side while greater values direct the camera more t
+    oward the vehicles lefthand side. This file can be changed incrementally with this function
+    or manually in the file system.
+    '''
+    def calibrate_pan(self, degree):
+        new_pan_cali = self.pan_offset + degree
+        self.db.set('pan_offset', new_pan_cali)
+        self.pan_offset = new_pan_cali
+
+
+    '''
+    Sends the panning input at maximum or minimum value if parameter entered exceeds maximum or 
+    minimum pan value.
+    '''
+    def limit_pan(self, degree):
+        if degree > 180:
+            degree = 180
+        if degree < 0:
+           degree = 0
+        return degree
+    
+
+    '''
+    Controls the til movement of the camera. Usable inputs are [0,150]. 0 refers to 
+    the vehicles front while 150 refers to the camerafacing upwards.
+    '''
+    def tilt(self, degree):
+        self.tilt_servo.write(self.limit_tilt(degree))
+
+    '''
+    Writes tilt offset to a config file within current directory, lower offset directs the camera
+    more towards the ground while higher offset directs the camera more vertical. This file can
+    be changed incrementally with this function or manually in the files system.
+    '''
+    def calibrate_tilt(self, degree):
```

Comment:
It would look cleaner if limit_tilt and limit pan were combined into one function, validate_degree(...) for an example.

Reply:
I agree, but I don't want the tilt to output 180 degrees, it may break the servo when it moves back that far.

Reply:
So don't do that, but still make it one function.